src/ua_timer.c

```diff
@@ -112,10 +114,30 @@ UA_Timer_addRepeatedCallback(UA_Timer *t, UA_ApplicationCallback callback,
     if(interval == 0)
         return UA_STATUSCODE_BADINTERNALERROR;
 
-    UA_DateTime nextTime = UA_DateTime_nowMonotonic() + (UA_DateTime)interval;
+    UA_DateTime currentTime = UA_DateTime_nowMonotonic();
+    UA_DateTime nextTime;
+    if(baseTime <= 0.0) {
+        nextTime = currentTime + (UA_DateTime)interval;
+    }
+    else {
+        if(baseTime > (currentTime + UA_DATETIME_SEC)) {
```

Comment:
The timer uses UA_DateTime_nowMonototic.
So it should be impossible travel back behind the baseTime.